store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -508,4 +528,36 @@ public static void addAccountToLogContextByAuthToken(Provisioning prov, AuthToke
         if (aid != null && !aid.equals(id))
             AccountUtil.addAccountToLogContext(prov, aid, ZimbraLog.C_ANAME, ZimbraLog.C_AID, null);
     }
+
+    public static AuthToken validateAndCreateNewJwtToken(String jwt, Account acct, Provisioning prov) throws ServiceException {
+        if (StringUtil.isNullOrEmpty(jwt)) {
+            throw ServiceException.AUTH_REQUIRED();
+        }
+        String tokenKey = acct.getJWTKey();
```

Comment:
can Storing JWT secret key on account level satisfy the requirement of multiple user sessions requirement? 
In case multiple users logged in from the same browser how will server understand which secret key should be used to validate the token. 

Is there any change in the requirement from client side? 

@shrikant-prasad  already raised this point.